vyper/context/types/indexable/mapping.py

```diff
@@ -28,29 +27,32 @@ def get_index_type(self, node):
 
 
 class MappingPrimitive(BasePrimitive):
-    _id = "map"
+    _id = "HashMap"
     _valid_literal = ()
 
     @classmethod
     def from_annotation(
         cls,
-        node: Union[vy_ast.Name, vy_ast.Call],
+        node: Union[vy_ast.Name, vy_ast.Call, vy_ast.Subscript],
         location: DataLocation = DataLocation.UNSET,
         is_constant: bool = False,
         is_public: bool = False,
     ) -> MappingDefinition:
-        if isinstance(node, vy_ast.Name):
-            raise CompilerPanic("Node must be a call")
+        if not isinstance(node, vy_ast.Subscript):
+            raise CompilerPanic("Node must be a subscript")
+        if not isinstance(node.slice.value, vy_ast.Tuple):
+            raise CompilerPanic("Mapping must be a subscript with a Tuple Index")
```

Comment:
This and the above shouldn't be `CompilerPanic` - they're easily reachable by the user.  I think `StructureException` is the most fitting.

Reply:
Also, reference to `Tuple` isn't clear imo.  The average user has no idea that the 2 numbers are represented as a tuple in the AST. It should just be something like "subscript with 2 values"

Reply:
See if you like how I solved this in a887410

Reply:
It's good!